penguin/tools/image_tools.py

```
@@ -0,0 +1,73 @@
+from PIL import Image
+
+from penguin.utils.path_utils import enforce_allowed_path
+
+
+SUPPORTED_IMAGE_FORMATS = {"JPEG", "PNG", "GIF", "WEBP", "BMP", "TIFF"}
+
+
+class ReadImageTool:
+    """Validate an image path and return model-visible artifact metadata."""
+
+    def execute(
```

Ruff (0.15.12)

[warning] Too many return statements (7 > 6)

(PLR0911)